pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -390,6 +398,33 @@ class Timestamp(_Timestamp):
             # User passed tzinfo instead of tz; avoid silently ignoring
             tz, tzinfo = tzinfo, None
 
+        # Allow fold only for unambiguous input
+        if fold is not None:
+            if fold not in [0, 1]:
+                raise ValueError(
+                    "Valid values for the fold argument are None, 0, or 1."
+                )
+
+            if (ts_input is not _no_input and (
+                    is_integer_object(ts_input) or is_float_object(ts_input) or
+                    is_datetime64_object(ts_input) or isinstance(ts_input, str)
+                    or getattr(ts_input, 'tzinfo', None) is not None)):
```

Comment:
if the only valid ts_input is a tznaive datetime, why not just check for that instead of all these separate checks?

Reply:
Done, thanks.